### PR TITLE
Deprecate :ruby - part 3

### PR DESCRIPTION
**Pull request by commitay**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/weechat.rb [hunk 1]
```diff
@@ -25,7 +25,7 @@ class Weechat < Formula
   depends_on "aspell" => :optional
   depends_on "lua" => :optional
   depends_on "python" => :optional
-  depends_on :ruby => ["2.1", :optional]
+  depends_on "ruby" => :optional
   depends_on :perl => ["5.3", :optional]
```
Comment (commitay) [inline Formula/weechat.rb]: `:perl is deprecated. Usage should be "perl"`

https://github.com/Homebrew/homebrew-core/pull/22703

cc @ilovezfs

File: Formula/libstfl.rb [hunk 2]
```diff
@@ -15,7 +15,7 @@ class Libstfl < Formula
   option "without-perl", "Build without Perl support"
   option "without-python", "Build without Python 2 support"
 
-  depends_on :ruby => ["1.8", :recommended]
+  depends_on "ruby" => :recommended if MacOS.version <= :mountain_lion
```
Comment (ilovezfs) [inline Formula/libstfl.rb]: needs an explicit ` option "without-ruby"` so there's still a corresponding option on > Mountain Lion

File: Formula/weechat.rb [hunk 3]
```diff
@@ -25,7 +25,7 @@ class Weechat < Formula
   depends_on "aspell" => :optional
   depends_on "lua" => :optional
   depends_on "python" => :optional
-  depends_on :ruby => ["2.1", :optional]
+  depends_on "ruby" => :optional
```
Comment (ilovezfs) [inline Formula/weechat.rb]: think we need a `if MacOS.version <= :sierra` here

File: Formula/weechat.rb (hunk 1 above)
Comment (ilovezfs) [reply Formula/weechat.rb]: yup no need to fix it here

File: Formula/xapian.rb [hunk 4]
```diff
@@ -20,7 +20,7 @@ class Xapian < Formula
   deprecated_option "php" => "with-php"
   deprecated_option "ruby" => "with-ruby"
 
-  depends_on :ruby => ["2.1", :optional]
+  depends_on "ruby" => :optional
```
Comment (ilovezfs) [inline Formula/xapian.rb]: think we need a `if MacOS.version <= :sierra` here

File: Formula/mkvtoolnix.rb [hunk 5]
```diff
@@ -3,7 +3,7 @@ class Mkvtoolnix < Formula
   homepage "https://www.bunkus.org/videotools/mkvtoolnix/"
   url "https://mkvtoolnix.download/sources/mkvtoolnix-19.0.0.tar.xz"
   sha256 "4b4af84cfdf3062f8ca4ab3c37854be7cd9a71bcf5c46b8ca805a101d414b6a1"
-  revision 1
+  revision 2
```
Comment (ilovezfs) [inline Formula/mkvtoolnix.rb]: don't need a revision bump for a build-time dependency

**Comment by ilovezfs** [comment]
Thanks @commitay!

**Pull request merged**